pennylane/ops/functions/equal.py

```diff
@@ -168,6 +169,38 @@ def _equal(
     raise NotImplementedError(f"Comparison of {type(op1)} and {type(op2)} not implemented")
 
 
+@_equal.register
+def _equal_circuit(
+    op1: qml.tape.QuantumScript,
+    op2: qml.tape.QuantumScript,
+    check_interface=True,
+    check_trainability=True,
+    rtol=1e-5,
+    atol=1e-9,
+):
+    # operations
+    if len(op1.operations) != len(op2.operations):
+        return False
+    for comparands in itertools.zip_longest(op1.operations, op2.operations):
```

Comment:
Why has `itertools.zip_longest` been used here? If you've already validated that the length operations is the same for both scripts, then you can just use the vanilla python `zip`

```suggestion
    for comparands in zip(op1.operations, op2.operations):
```

Reply:
fixed in 6a4d2097aed42790c12ca6420a7c20381c45b8d6

Reply:
feel free to resolve this @mudit2812 :)